### ISSUE TITLE
Glyph .notdef embedded even if unused

### ISSUE THREAD

**Issue by RobtAll** [wspr/fontspec#281]
This might be an issue with luaotfload, rather than fontspec. But I cannot decipher that code.

Most Open Type fonts have a visible glyph (often a crossed box) for the .notdef slot. A few fonts simply leave .notdef as a blank glyph, either the width of a space or the width of em.

Issue: If the font has a visible .notdef glyph, it is placed in the PDF, whether or not it is needed. MWE:

    % !TeX program = LuaLaTeX
    % !TeX encoding = UTF-8
    \documentclass{article}
    \usepackage{fontspec}
    \setmainfont{Linux Libertine O}
    \begin{document}
    \thispagestyle{empty}
    a\par
    \end{document}

If you extract the font from the PDF using FontForge, you will see that it contains exactly two glyphs: The letter a, and the .notdef box, even though there is no undefined character in the file. Font "Linux Libertine O" has a crossed-box .notdef glyph.

For comparison, comment-out \setmainfont, so that the default "Latin Modern Roman" is used instead. This font has a blank .notdef. When you extract the font from the PDF, there is no .notdef glyph there, not even a blank.

In other words: The presence of a non-blank .notdef causes that glyph to be embedded in the PDF, even if it is unused and no characters refer to it.

Why this might be a problem: In PDF/X and PDF/A, any reference to .notdef is prohibited, because the PDF is not allowed to have missing characters. Even though there is no reference, the presence of .notdef may mislead a reviewer into thinking that it is used.

**Comment by wspr** [comment]
Hi Robert; thanks for the detailed report. This might not even be an issue with luaotfload; if anything I suspect it’s the inbuild pdf distiller in LuaTeX. Unfortunately I can’t track down errors in that code; I would recommend asking on a LuaTeX mailing list.

**Comment by RobtAll** [comment]
All right, if it's that deep, then I'll look in that direction.
